File: D3/adv2/src/main.rs
use std::fs;

fn main() {
    let file = fs::read_to_string("/home/operaho/Advent/D3/adv.txt").unwrap();
    let mut c = 1;
    let mut v = Vec::new();
    let mut v0 = Vec::new();
    let mut v1 = Vec::new();

    for i in file.lines() {
        v.push(i);
        v0.push(i);
    }

    while c <= 12 {
        v1.clear();

        for j in &v0 {
            v1.push(&j[c-1..c]);
        }

        let f0 = v1.iter().filter(|&n| *n == "0").count();
        let f1 = v1.iter().filter(|&n| *n == "1").count();

        if f0 >= f1 { // change the order of the operator to get the other count
            for k in &v {
                if k.starts_with("0") {
                    let pos: usize = v.iter().position(|x| *&x == k).unwrap();

                    v.remove(pos); // wait for it
                }
            }
        } else {
            for k in &v {
                if k.starts_with("1") {
                    let pos: usize = v.iter().position(|x| *&x == k).unwrap();

                    v.remove(pos); // wait for it
                }
            }
        }
        
        c += 1;
    }

    println!("{:?}", v);
}

File: D3/adv2/main.py
file = open("/home/operaho/Advent/D3/adv.txt", "r")
f, l, out, c = file.read().splitlines(), [], [], 0

while len(f) > 1:
	l.clear()
	out.clear()

	if c == 12:
		c = c*0

	for i in f:
		l.append(i[c])

	l0 = l.count("0")
	l1 = l.count("1")

	if l0 > l1:
		for j in f:
			if j[c] == "1": # Change this to find the other binary
				out.append(j)
		f = out[:]
	if l1 > l0:
		for j in f:
			if j[c] == "0": # Change this to find the other binary
				out.append(j)
		f = out[:]
	if l0 == l1:
		for j in f:
			if j[c] == "0": # Change this to find the other binary
				out.append(j)
		f = out[:]

	c = c + 1

print(f)
